File: src/app/services/jaw.service.ts
import { HttpClient } from '@angular/common/http';
import { Injectable } from '@angular/core';
import { Observable } from 'rxjs';
import { tap } from 'rxjs/operators';
import { Tooth } from '../models/patient';

@Injectable({
  providedIn: 'root'
})
export class JawService {

  getUrl = 'http://localhost:8080/api/patient';
  constructor(private http: HttpClient) {}

  getJaw(patientId: string): Observable<Tooth[]> {
    return this.http.get<Tooth[]>(`${this.getUrl}/${patientId}/jaw`).pipe(tap((res) => {
      console.log(res[0])
    }));
  }

  updateTooth(patientId: string, toothId: string, data: Tooth) {
    return this.http.put(`${this.getUrl}/${patientId}/jaw/${toothId}`, data)
  }
}

File: src/main/java/com/dentistapp/dentistappdevelop/service/JawService.java
package com.dentistapp.dentistappdevelop.service;


import com.dentistapp.dentistappdevelop.model.Jaw;
import com.dentistapp.dentistappdevelop.model.Patient;
import com.dentistapp.dentistappdevelop.model.Recipe;
import com.dentistapp.dentistappdevelop.model.Tooth;

import java.util.List;


public interface JawService {

    public Jaw findJawByPatientId(String patientId);
    public Jaw update(String patientId, Jaw jaw);
    public Tooth updateTooth(String patientId, String toothId, Tooth tooth);

}
